pkg/auth/dependency/webapp/redirect.go

```diff
@@ -112,3 +116,14 @@ func MakeURLWithQuery(u *url.URL, query url.Values) string {
 	}
 	return fmt.Sprintf("?%s", q.Encode())
 }
+
+func NewURLWithPathAndQuery(path string, query url.Values) string {
```

Comment:
This function does not preserve Auth UI query parameters like `redirect_uri` and `ui_locales`. In the callback endpoint, we need to restore the persisted encoded query into `r` as well as the `x_sid`.